### ISSUE TITLE
async await broken in 1.6-beta.16 (Windows x64)

### ISSUE THREAD

**Issue by jakobrosenberg** [meteor/meteor#9002]
On Windows 10 x64 the following code works in a new Meteor project v1.5.1, but not v1.6-beta.16.

```
const Cars = new Mongo.Collection("cars")

const fakeDelay = async function () {
  return new Promise((resolve, reject) => {
    setTimeout(() => { resolve('we waited long enough') }, 1000)
  })
}

Meteor.methods({
  async findcar() {
    await fakeDelay()    
    return Cars.findOne()
  }
})
```

In the 1.6-beta.16 the code above produces the following error ``Exception while invoking method 'findcar' Error: Can't wait without a fiber``

**Comment by formspoint** [comment]
wait -- we have to add an 'await' to the front of the function call if calling an async function?  I thought we could do the following to make the async call:

// module_a.js => server-side
```javascript
 // => don't  hold things up, so make async
thing.getExistingOrNewThingAsync = async userId => {
  let things = null;
  if (await thing.expiredOrNotExistsThing(userId)) {
    // adding await to sync function
    things = await tokens.getNewThing(userId); // => regular sync function
    return things;
  }
  else {
    // adding await to sync function
    things = await tokens.getThing(userId); // => regular sync function
    return things;
  }
}
```

```javascript
// login handler => server-side
import { thing } from 'imports/server/stuff/module_a';
Accounts.onLogin(user => {
  const userId = user && user.user && user.user._id || '';
  thing.getExistingOrNewThingAsync(userId); // => no await here, but is called async
});
```

Please correct me here!

**Comment by aadamsx** [comment]
wait -- we have to add an 'await' to the front of the function call if calling an async function?  I thought we could do the following to make the async call:

// module_a.js => server-side
```javascript
 // => don't  hold things up, so make async
thing.getExistingOrNewThingAsync = async userId => {
  let things = null;
  if (await thing.expiredOrNotExistsThing(userId)) {
    // adding await to sync function
    things = await tokens.getNewThing(userId); // => regular sync function
    return things;
  }
  else {
    // adding await to sync function
    things = await tokens.getThing(userId); // => regular sync function
    return things;
  }
}
```

```javascript
// login handler => server-side
import { thing } from 'imports/server/stuff/module_a';
Accounts.onLogin(user => {
  const userId = user && user.user && user.user._id || '';
  thing.getExistingOrNewThingAsync(userId); // => no await here, but is called async
});
```

Please correct me here!

**Comment by klaussner** [comment]
A bit more debugging information: the error is thrown in the `Cars.findOne` call if an expression has been `await`ed before in the method. After the `await`, `Promise.Fiber.current` is `undefined`. Full stack trace:

```
Exception while invoking method 'findcar' Error: Can't wait without a fiber
    at Function.wait ([...]/dev_bundle/server-lib/node_modules/fibers/future.js:159:9)
    at Future.wait ([...]/dev_bundle/server-lib/node_modules/fibers/future.js:448:10)
    at SynchronousCursor._nextObject (packages/mongo/mongo_driver.js:1018:47)
    at SynchronousCursor.forEach (packages/mongo/mongo_driver.js:1052:22)
    at SynchronousCursor.map (packages/mongo/mongo_driver.js:1062:10)
    at SynchronousCursor.fetch (packages/mongo/mongo_driver.js:1086:17)
    at Cursor.(anonymous function) [as fetch] (packages/mongo/mongo_driver.js:876:44)
    at MongoConnection.findOne (packages/mongo/mongo_driver.js:783:56)
    at Mongo.Collection.findOne (packages/mongo/collection.js:334:29)
    at Promise.asyncApply (server/main.js:25:17)
```

@aadamsx If you don't `await` the promise returned by `getExistingOrNewThingAsync`, you are just ignoring the value it's resolved with. But that's off-topic for this issue. 🙂

**Comment by aadamsx** [comment]
@klaussner yep, in my case, in this situation, I don't care about the return value.

**Comment by benjamn** [comment]
Can we get a reproduction of this? Happy to take a look.

**Comment by klaussner** [comment]
@benjamn https://github.com/klaussner/meteor-issues/tree/9002 (with @jakobrosenberg's example)

**Comment by benjamn** [comment]
This should be fixed if you run `meteor update --release 1.6-beta.18` in your app. Thanks for reporting the problem!
